src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -111,6 +111,11 @@ async function put_bucket_lifecycle(req) {
         if (rule.Status?.length !== 1) {
             dbg.error('Rule should have status', rule);
             throw new S3Error(S3Error.InvalidArgument);
+        } else if (rule.Status?.length === 1) {
+            if (rule.Status[0] !== "Enabled" && rule.Status[0] !== "Disabled") {
```

Comment:
Why do you access it as an array? I thought we were expecting a string...
Please move "Enabled" and "Disabled" to an enum (you can use the s3_constant file).

Reply:
You can check what you get with log printings, you can add `typeof <variable-name>` to check it.

Reply:
It is of `object` type.